pkg/kvstore/allocator/cache.go

```diff
@@ -200,6 +203,7 @@ func (c *cache) start(a *Allocator) waitChan {
 						}
 
 						delete(c.nextCache, id)
+						a.idPool.Insert(id)
```

Comment:
After the initial list is completed, the call to 'FinishRefresh' causes the 'current ID cache' to point to the 'next ID cache' so that the Insert and Remove operations continue to be done on the 'current ID cache'. The pointers to the caches will only differ during the list operation.